app/components/files/file.tsx

```diff
@@ -66,74 +77,136 @@ const File = ({
         }
     }, [index]);
 
+    const handleOnOptionsPress = useCallback(() => {
+        if (onOptionsPress) {
+            onOptionsPress(index);
+        }
+    }, [index, onOptionsPress]);
+
+    const renderOptionsButton = useCallback(() => {
```

Comment:
You are always calling this function, not passing as a callback, and nobody depends on it, so there is no need for `useCallback`.